components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -7172,6 +7192,70 @@ public Application getApplicationById(int applicationId) throws APIManagementExc
         return application;
     }
 
+    /** Retrieves the Application which is corresponding to the given UUID String
+     * 
+     * @param uuid UUID of Application
+     * @return
+     * @throws APIManagementException
+     */
+    public Application getApplicationByUUID(String uuid) throws APIManagementException {
+        Connection connection = null;
+        PreparedStatement prepStmt = null;
+        ResultSet rs = null;
+
+        Application application = null;
+        try {
+            connection = APIMgtDBUtil.getConnection();
+
+            String query = "SELECT " +
+                    "APP.APPLICATION_ID," +
+                    "APP.NAME," +
+                    "APP.SUBSCRIBER_ID," +
+                    "APP.APPLICATION_TIER," +
+                    "APP.CALLBACK_URL," +
+                    "APP.DESCRIPTION, " +
+                    "APP.SUBSCRIBER_ID,"+
+                    "APP.APPLICATION_STATUS, " +
+                    "APP.UUID," +
+                    "SUB.USER_ID " +
+                    "FROM " +
+                    "AM_SUBSCRIBER SUB," +
+                    "AM_APPLICATION APP " +
+                    "WHERE APP.UUID = ? " +
+                    "AND APP.SUBSCRIBER_ID = SUB.SUBSCRIBER_ID";
+
+            prepStmt = connection.prepareStatement(query);
+            prepStmt.setString(1, uuid);
+
+            rs = prepStmt.executeQuery();
+
+            while (rs.next()) {
```

Comment:
If you are only expecting at most 1 result for the above statement you can use an if condition here instead of a while. No need to create a loop and break on first execution.